appservice/src/createAppService/CustomLocationListStep.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtLocation, IAzureQuickPickItem, IAzureQuickPickOptions, LocationListStep, parseError } from 'vscode-azureextensionui';
+import { localize } from '../localize';
+import { createResourceGraphClient } from '../utils/azureClients';
+import { CustomLocation, IAppServiceWizardContext } from './IAppServiceWizardContext';
+
+export class CustomLocationListStep<T extends IAppServiceWizardContext> extends LocationListStep<T> {
+    public async prompt(wizardContext: T): Promise<void> {
+        const options: IAzureQuickPickOptions = { placeHolder: localize('selectLocation', 'Select a location for new resources.'), enableGrouping: true };
+        const result: AzExtLocation | CustomLocation = (await wizardContext.ui.showQuickPick(this.getCustomQuickPicks(wizardContext), options)).data;
+        if ('kubeEnvironment' in result) {
+            wizardContext.customLocation = result;
+            // For any resources other than the app, we still need a non-custom location, so we'll use the kubeEnvironment's location
+            await LocationListStep.setLocation(wizardContext, result.kubeEnvironment.location);
+            // Plan has very little effect when a custom location is used, so default this info instead of prompting
+            wizardContext.newPlanName = await wizardContext.relatedNameTask;
+            wizardContext.newPlanSku = { name: 'K1', tier: 'Kubernetes', size: 'K1', family: 'K', capacity: 1 };
+        } else {
+            wizardContext.location = result;
+        }
+    }
+
+    protected async getCustomQuickPicks(wizardContext: T): Promise<IAzureQuickPickItem<AzExtLocation | CustomLocation>[]> {
```

Comment:
Why call this `getCustomQuickPicks` when this is already the `CustomLocationListStep`?

Reply:
To differentiate between `super.getQuickPicks` which is getting the other ones